providers/base/tests/test_edid_cycle.py

```diff
@@ -0,0 +1,164 @@
+"""This module provides test cases for the edid_cycle module."""
+import unittest
+from unittest.mock import patch, call, Mock
+
+from bin import edid_cycle
+
+
+class ZapperEdidCycleTests(unittest.TestCase):
+    """This class provides test cases for the edid_cycle module."""
+
+    @patch("time.sleep", new=Mock)
+    @patch("bin.edid_cycle.zapper_run", new=Mock)
+    @patch("builtins.open")
+    @patch("os.getenv")
+    @patch("subprocess.check_output")
+    def test_discover_video_output_device(
+        self, mock_check, mock_getenv, mock_open
+    ):
+        """
+        Check if the function automatically discover the port
+        under test with output given by randr in different conditions.
+        """
+        disconnected_output = {"x11": "Monitors: 0", "wayland": ""}
+
+        connected_output = {
+            "x11": (
+                "Monitors: 1\n"
+                " 0: +HDMI-1 1920/576x1080/324+800+1080 HDMI-1\n"
+            ),
+            "wayland": (
+                "x: 0, y: 0, scale: 1, rotation: normal, primary: yes\n"
+                "associated physical monitors:\n"
+                "    HDMI-1 TSB PI-KVM Video 0x88888800\n\n"
+                "HDMI-1 TSB PI-KVM Video\n"
+            ),
+        }
+
+        edid_cycle.EDID_FILES = ["1920x1080.edid"]
+
+        for env in ["x11", "wayland"]:
+            mock_getenv.return_value = env
+            mock_open.reset_mock()
+
+            # Happy Case
+            mock_check.side_effect = [
+                disconnected_output[env],
+                connected_output[env],
+            ]
+
+            port = edid_cycle.discover_video_output_device("zapper-ip")
+            assert port == "HDMI-1"
+
+            # Failure
+            mock_check.side_effect = None
+            mock_check.return_value = connected_output[env]
+            with self.assertRaises(IOError):
+                edid_cycle.discover_video_output_device("zapper-ip")
+
+    @patch("time.sleep", new=Mock)
+    @patch("bin.edid_cycle.zapper_run", new=Mock)
+    @patch("builtins.open")
+    @patch("os.getenv")
+    @patch("subprocess.check_output")
+    def test_test_edid(self, mock_check, mock_getenv, mock_open):
```

Comment:
Both this and `test_discover_video_output_device` is way too smart, split x11 and wayland, and split happy and non-happy paths. Any time one of these tests fails you have to dig into the log to know what is wrong and can't rely on the name!